include/multigrid_pclomp/multi_voxel_grid_covariance_omp.h

```diff
@@ -252,7 +314,65 @@ class MultiVoxelGridCovariance : public pcl::VoxelGrid<PointT> {
   // Return the string indices of currently loaded map pieces
   std::vector<std::string> getCurrentMapIDs() const;
 
+  void setThreadNum(int thread_num) {
+    if(thread_num <= 0) {
+      thread_num_ = 1;
+    }
+
+    thread_num_ = thread_num;
+    thread_futs_.resize(thread_num_);
```

Comment:
Just to confirm, is sync() not necessary before resizing?
I think that there might be a possibility of calling setThreadNum immediately after calling setInputCloudAndFilter.

Reply:
@SakodaShintaro Thanks. I added a sync in the begginning of setThreadNum.